libpyvinyl/Parameters/Parameter.py

```diff
@@ -8,126 +8,168 @@ class Parameter(AbstractBaseClass):
     """
     Description of a single parameter
     """
+
     def __init__(self, name, unit=None, comment=None):
         """
         Creates parameter with given name, optionally unit and comment
         """
         self.name = name
         self.unit = unit
         self.comment = comment
-        self.value = None
-        self.legal_intervals = []
-        self.illegal_intervals = []
+        self._value = None
+        self.intervals = []
+        self._intervals_are_legal = None
         self.options = []
+        self._options_are_legal = None
 
     @classmethod
     def from_dict(cls, param_dict):
-        param = cls(param_dict['name'], param_dict['unit'],
-                    param_dict['comment'])
+        param = cls(param_dict["name"], param_dict["unit"], param_dict["comment"])
         for key in param_dict:
             param.__dict__[key] = param_dict[key]
         return param
 
-    def add_legal_interval(self, min_value, max_value):
+    def add_interval(self, min_value, max_value, intervals_are_legal):
```

Comment:
With the new way of having a global legal / illegal bool, I think this is a good way to input the information that is clear to the user.